spec/models/link_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'helper'
+
+describe Link do
+  describe 'URL validation' do
+    subject { described_class.new url: link, shortlink: 'g' }
+
+    before { subject.validate }
+
+    describe 'http links' do
+      let(:link) { 'http://www.google.com' }
+
+      it { is_expected.to be_valid }
+    end
+
+    describe 'https links' do
+      let(:link) { 'https://www.google.com' }
+
+      it { is_expected.to be_valid }
+    end
+
+    describe 'javascript links' do
```

Comment:
Agreeing with Jens about verifying whether this is a sufficiently exhaustive validator. But that can wait for pull request two, this is definitely better than nothing. :+1: